### PR TITLE
Include stack in Cloud Code

### PR DESCRIPTION
**Pull request by dblythy**
At the moment in Cloud Code, if an error is thrown from a syntax / JS error, such as:

```
Parse.Cloud.define("testFunction", async (request) => {
    yolo
});
```
The following error is shown:
```
error: Parse error:  yolo is not defined {"code":141,"stack":"Error: yolo is not defined
    at error (/node_modules/parse-server/lib/Routers/FunctionsRouter.js:121:16)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)}"
```

Which can make it a bit difficult sometimes to trace when your main.js is long.

This PR passes stack through if the error thrown is an `Error`, meaning that the following is logged, making it a bit easier to identify where the syntax error was.

```
error: Parse error:  yolo is not defined {"code":141,"stack":"ReferenceError: yolo is not defined
    at  /cloud/main.js:3:5
    at /node_modules/parse-server/lib/Routers/FunctionsRouter.js:199:16
    at processTicksAndRejections (internal/process/task_queues.js:93:5)"}
```

Sorry for submitting so many PRs!

### CODE REVIEW

**Comment by dplewis** [comment]
I was just thinking about this error handling.

**Comment by dblythy** [comment]
@dplewis would you have any suggestions / pointers as to how to get relevant trace if a Parse.Error is thrown? The solution in this PR only works if if a JS error is thrown.

 E.g:

```
Parse.Cloud.define("testFunction", async (request) => {
    let obj = new Parse.Object('Test');
    await obj.save();
    await obj.destroy();
    await obj.save();
});
```

Throws 
```
error: Parse error:  Object not found. {"code":101,"stack":"Error: Object not found.
    at /node_modules/parse-server/lib/Controllers/DatabaseController.js:563:23
    at processTicksAndRejections (internal/process/task_queues.js:93:5)"}
```

Which again, isn't overly helpful at identifying where the error is comes from. Obviously, you can wrap it in a try / catch block, but just thinking how we can make it as novice-friendly as possible.

**Comment by dplewis** [comment]
There was a conversation about this. https://github.com/parse-community/parse-server/issues/6205#issuecomment-553381958

One quick fix that comes to mind would be to remove the promise chaining that happens in the rest, restquery, restwrite, databaseController etc.

File: src/Routers/FunctionsRouter.js [hunk 1]
```diff
@@ -119,10 +119,11 @@ export class FunctionsRouter extends PromiseRouter {
         if (typeof message === 'string') {
           return reject(new Parse.Error(code, message));
         }
+        const error = new Parse.Error(code, message.message || message);
```
Comment (davimacedo) [inline src/Routers/FunctionsRouter.js:122]: Thanks for the PR! I'm only afraid of `message` being null/undefined for any reason. So I'd suggest `const error = new Parse.Error(code, message && message.message || message);`

**Review by davimacedo — APPROVED** [review]
LGTM!

**Pull request merged**